server/api_test.go

```diff
@@ -1,90 +1,89 @@
 package server_test
 
 import (
-	"context"
 	"net/http"
 	"net/http/httptest"
 	"strconv"
 	"testing"
 
-	"github.com/Hukyl/genesis-kma-school-entry/database"
 	"github.com/Hukyl/genesis-kma-school-entry/models"
 	"github.com/Hukyl/genesis-kma-school-entry/rate"
 	"github.com/Hukyl/genesis-kma-school-entry/server"
 	serverCfg "github.com/Hukyl/genesis-kma-school-entry/server/config"
 	"github.com/Hukyl/genesis-kma-school-entry/settings"
 	"github.com/gin-gonic/gin"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 )
 
-func TestEmptyContext(t *testing.T) {
-	engine := server.NewEngine(context.Background())
-	req, err := http.NewRequest("GET", "/rate", nil)
-	if err != nil {
-		t.Fatal(err)
-	}
-	rr := httptest.NewRecorder()
-	engine.ServeHTTP(rr, req)
-	assert.Equal(t, http.StatusInternalServerError, rr.Code)
+type mockRateFetcher struct {
+	mock.Mock
+}
+
+func (m *mockRateFetcher) FetchRate(from, to string) (rate.Rate, error) {
+	args := m.Called(from, to)
+	return args.Get(0).(rate.Rate), args.Error(1)
+}
+
+type mockUserRepository struct {
+	mock.Mock
+}
+
+func (m *mockUserRepository) FindAll() ([]models.User, error) {
+	args := m.Called()
+	return args.Get(0).([]models.User), args.Error(1)
+}
+
+func (m *mockUserRepository) Create(user *models.User) error {
+	args := m.Called(user)
+	return args.Error(0)
+}
+
+func (m *mockUserRepository) Exists(user *models.User) (bool, error) {
+	args := m.Called(user)
+	return args.Bool(0), args.Error(1)
 }
 
 func TestGetRate(t *testing.T) {
-	ctx := context.Background()
-	ctx = context.WithValue(ctx, settings.DebugKey, true)
-	apiClient := server.Client{
-		Config:      serverCfg.NewFromEnv(),
-		RateFetcher: rate.NewNBURateFetcher(),
-	}
-	ctx = context.WithValue(ctx, settings.APIClientKey, apiClient)
-	engine := server.NewEngine(ctx)
+	mockFetcher := new(mockRateFetcher)
+	mockedRate := rate.Rate{Rate: 27.5}
+	mockFetcher.On("FetchRate", "USD", "UAH").Return(mockedRate, nil)
+	engine := server.NewEngine(server.Client{
+		Config:      serverCfg.Config{Port: "8080"},
+		RateFetcher: mockFetcher,
+	})
 
-	// FIXME: tests should not depend on external services
-	req, err := http.NewRequest("GET", "/rate", nil)
-	if err != nil {
-		t.Fatal(err)
-	}
+	req := httptest.NewRequest(http.MethodGet, server.RatePath, nil)
```

Comment:
```suggestion
	req := httptest.NewRequest(http.MethodGet, "/rate", nil)
```

Краще буде здублювати path, поясню чому на можливому кейсі в уявній компанії:
1. Хтось випдаково замінив server.RatePath і якимось чудом прошов рев'ю або рев'ю в компанії відсутнє.
2. Тести пройшли оскільки залежать від значення змінної, проте контракт змінився.
3. FE або користувач користується старим контрактом і отримує помилку.

А якщо здублювати `/rate`, то тести заваляться і розробник зверне увагу що значення змінилось